app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+
+  def new
+    redirect_to home_path if current_user
+  end
+
+  def create
+    user = User.where(email: params[:email]).first
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to home_path, notice: "You are signed in, enjoy!"
+    else
+      flash[:error] = "Invalid email or password."
```

Comment:
Bootstrap 3 now uses `danger` instead of `error`. You can see the correct classes [here](http://getbootstrap.com/components/#alerts).